libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -754,6 +754,40 @@ string YulUtilFunctions::storageArrayIndexAccessFunction(ArrayType const& _type)
 	});
 }
 
+string YulUtilFunctions::memoryArrayIndexAccessFunction(ArrayType const& _type)
+{
+	string functionName = "memory_array_index_access_" + _type.identifier();
+	return m_functionCollector->createFunction(functionName, [&]() {
+		return Whiskers(R"(
+			function <functionName>(baseRef, index) -> addr {
+				if iszero(lt(index, <arrayLen>(baseRef))) {
+					invalid()
+				}
+
+				let offset := index
+				<?noByteArray>
+					offset := mul(index, <headSize>)
+				</noByteArray>
+				<?dynamicallySized>
+					offset := add(offset, 32)
+				</dynamicallySized>
+				addr := add(baseRef, offset)
+			}
+		)")
+		("functionName", functionName)
+		("arrayLen", arrayLengthFunction(_type))
+		("headSize", to_string(_type.memoryHeadSize()))
```

Comment:
Is this correct? Isn't the idea to multiply the index by the size of each element? If so I would imagine this should use the base type.
In case this is true /\, using `ArrayType::memoryStride()` would allow removing `<?noByteArray>`.

Reply:
Yep! https://github.com/ethereum/solidity/pull/7015#pullrequestreview-258934112

@marenz why didn't you include that change suggestion?

Reply:
I made the change locally but haven't pushed yet. I mark things as resolved as I fix them and then push.

Reply:
Changing it to use memorystride

Reply:
Ah, indeed, that seems like the perfect fit ;)